### DISCUSSION TITLE
Welcome to LSPosed Discussions!

### DISCUSSION THREAD

**Discussion by kotori2** [LSPosed/LSPosed#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by XspeedPL** [comment]
Hey, so like, why another fork?

Also out of curiosity, where does the new name come from?

**Reply by kotori2** [reply]
> Hey, so like, why another fork?

The og author refused lots of breaking changes that kills backwards compability.
Also the release in Magisk and EdXposed Manager is always not up to date. 

> Also out of curiosity, where does the new name come from?

~~old horny bitch~~ ←idk if I translate this correct
~~It was a joke at first and it just became official~~

**Reply by XspeedPL** [reply]
> The og author refused lots of breaking changes that kills backwards compability.
> Also the release in Magisk and EdXposed Manager is always not up to date.

That is true and I agree, but on the other hand the community is already too spread out.
I am afraid that introducing more flavors of Xposed may further confuse and turn away more people.

So in short, you want to drop backwards compatibility? Device compatibility or module compatibility?
Truth be told, I sometimes use old modules, like Greenify or MinMinGuard.

> ~old horny bitch~ ←idk if I translate this correct
> ~It was a joke at first and it just became official~

I see :P
Some people spell Xposed as "XPosed" for some goddamn reason, so I was wondering if the casing was intentional here.

**Reply by kotori2** [reply]
> I am afraid that introducing more flavors of Xposed may further confuse and turn away more people.

Yes. To keep evolving, it's great for us to have some competitors, or "fork" in open source project. 

> So in short, you want to drop backwards compatibility?

Something like, dropping support for old EdXposed Installer or introducing new Xposed API. 
Also some behaviour that developers shouldn't do, such as download data (blocking) in SystemUI, or accessing shared_prefs from their own private data folder. 
If we made an exception for these behaviour, ofc it will work but leading to security issues or even slow down the whole system.

**Reply by XspeedPL** [reply]
Sounds great, you got me convinced

**Comment by C3C0** [comment]
Hi. I'm a dev of GravityBox module. Looking forward to where this will go. Just pushed one commit that was hanging there for some time in og repo.

**Reply by yujincheng08** [reply]
Hi! We introduced Module Scope metadata to make users select scopes easily for modules. We would like GravityBox to do this tiny change: https://github.com/LSPosed/LSPosed/wiki/Module-Scope

**Reply by XspeedPL** [reply]
It uses an array resource, can we assume it will support alternative configurations?
Like having values.xml and values-v21.xml

**Reply by yujincheng08** [reply]
@XspeedPL Yes! That's a feature! You can declare different scopes for different SDK versions.

**Comment by kotori2** [comment]
What about... Open a new thread

**Reply by yujincheng08** [reply]
Just for convenience 😜

**Comment by Neubulae** [comment]
~~Give a proper back-formation of LSPosed to further distinguish your fork~~jk, anyway, would you elaborate a bit on why you decided to fork out? e.g. what kinds of changes are those that broke backward-compatibility™? and why you deem them necessary? How different is this from EdXp? and what kind of difference would end users notice?
Do you consider yourselves as sole competitors or this is just a *complementary* fork? Tho being the latter also means overshadowing the upstream, therefore making no difference between 2 choices…
Anyway, I hope you a glorious success! May you receive your acclaim with your steps onward!

**Reply by yujincheng08** [reply]
Just see issues with `[TODO]` prefix.

**Comment by Mikanoshi** [comment]
I wish someone created a fork called IDCASSNPosed as in "I don't care about stupid Safety Net" with direct code injection to zygote before the fork. You know, for best performance and system-wide framework hooks support, which are not compatible with LSPosed's activation scope feature.

**Comment by datsoy** [comment]
![Screenshot_20210208-105633](https://user-images.githubusercontent.com/67457715/107170950-98226c80-69fc-11eb-8900-62e7708d09dc.jpg)

Adding something like this would be great when a new app is intalled.

**Comment by qiuye1120** [comment]
你好

**Comment by nxsz379** [comment]
是否可以通过adb shell 勾选lsp模块作用域
